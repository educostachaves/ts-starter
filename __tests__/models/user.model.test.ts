import pool from '../../src/core/db';
import UserModel from '../../src/models/user.model';

function mockResult() {
  return {
    rows: [
    {
      id: 2,
      name: 'Eduardo Chaves',
      email: 'eduardo.chaves@ibm.com',
      created_on: '2019-09-26T09:37:57.707Z',
    },
    {
      id: 3,
      name: 'SalomÃ£o',
      email: 'raibersa@br.ibm.com',
      created_on: '2019-09-26T09:50:46.146Z',
    },
    {
      id: 4,
      name: 'Diogo Tocci',
      email: 'dtocci@br.ibm.com',
      created_on: '2019-09-26T09:51:14.602Z',
    },
    ],
  };
}

describe('UserModel', () => {
  describe('When List method is called', () => {
    it('should return a list of users', async () => {
      let query;
      pool.query = jest
        .fn()
        .mockImplementation((mockQuery) => {
          query = mockQuery;
          return mockResult();
        });

      const result = await UserModel.list();

      expect(query).toMatchSnapshot();
      expect(result.length).toBe(3);
    });

    it('should return a list of users', async () => {
      pool.query = jest
        .fn()
        .mockImplementation(() => {
          return {};
        });

      const result = await UserModel.list();

      expect(result.length).toBe(0);
    });

    it('should return status false when an exception happens', async () => {
      try {
        pool.query = jest
        .fn()
        .mockImplementation(() => {
          throw new Error('Mock Error');
        });

        const result = await UserModel.list();
      } catch (err) {
        expect(err.message).toBe('Mock Error');
      }
    });
  });
});
