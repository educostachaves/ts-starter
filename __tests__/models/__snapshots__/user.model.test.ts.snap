// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModel When List method is called should return a list of users 1`] = `
"
      SELECT * FROM users;
    "
`;
